### ISSUE TITLE
Codec quantization support

### ISSUE THREAD

**Issue by elnaske** [espnet/espnet#6326]
I noticed that only certain tasks support model quantization for inference. I've been experimenting with quantizing neural codecs, so I ported quantization functionality to `gan_codec_inference`. I've manually tested running inference through the command line as well as the `AudioCoding` class in Python using the public checkpoints for SoundStream, EnCodec, and DAC. While it's been working well so far, there are two things I'm unsure about with respect to the implementation:

1. Some of the models use [weight norms](https://docs.pytorch.org/docs/stable/generated/torch.nn.utils.parametrizations.weight_norm.html), which prevents PyTorch from quantizing the layers they're applied to. Given that the point of weight normalization is to speed up convergence during training, it should be safe to remove them for inference, right?
2. The `qconfig_spec` argument of `torch.quantization.quantize_dynamic` specifies which layers are quantized. If it is `None`, all layers are quantized. However, within the tasks that currently support quantization, there is no option to pass `None` as the argument for `qconfig_spec`, since the `quantize_modules` parameter requires you to explicitly define the names of the layers to be quantized. Is there a reason for this design decision? I think it might be nice to give people the option to implicitly quantize all layers.

I'm interested in hearing how these issues should be handled, and if this sounds like a useful feature, I can run some formal tests and open a pull request.

**Comment by sw005320** [comment]
Thanks for your reports.
They make sense to me.
@ftshijt, can you share your thoughts?

**Comment by elnaske** [comment]
Since there was no existing test script for codec inference, I opened a pull request that adds some basic unit tests that I'd need to confirm the functionality of my implementation. Please see #6332 .

**Comment by sw005320** [comment]
Thanks a lot!

**Comment by ftshijt** [comment]
For the first question, yes, you are correct. For inference purposes, that can be removed.

For the second, I do not think we have worked on related quantization (on model parameters) for codec yet, while the most usages for now are for other tasks. If you would like to have a check and implementation on that aspect, it would be super!

**Comment by ftshijt** [comment]
> Since there was no existing test script for codec inference, I opened a pull request that adds some basic unit tests that I'd need to confirm the functionality of my implementation. Please see [#6332](https://github.com/espnet/espnet/pull/6332) .

Many thanks! I've checked the PR and it looks good as a start point. Thanks for your contribution.

**Comment by elnaske** [comment]
@ftshijt Thanks for your review and for answering my questions. Since all the tests passed, feel free to merge that PR. I'll open a new one once the quantization implementation is ready.
